data/users.js

```diff
@@ -235,10 +235,13 @@ const exportMethod = {
       throw "Daycare could not be found or removed";
     }
 
-    const newId = daycareId.toString();
-    const favDaycare = await this.getFavDayCare(newId);
+    // The below is causing some type of race condition when you try to favorite and unfavorite a few times
+    // Given that nothing depends on the return for this, I think it can be removed since commenting it out fixes the issue
 
-    return favDaycare;
+    // const newId = daycareId.toString();
+    // const favDaycare = await this.getFavDayCare(newId);
```

Comment:
@witcherprince wanted to mention this so you saw. The fetch after favoriting a daycare here was causing exceptions. It seems like it wasnt waiting on the return for this function call. Given that the return here isnt used I think we can just remove. If you uncomment the code and favorite / unfavorite a daycare a few times youll see the crash Im talking about. Let me know if any issues w/ this.

Reply:
just to clarify, removing these two lines fixes the issue and favoriting functionality works w/o any exceptions